src/models/room.ts

```diff
@@ -914,12 +914,38 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.myUserId;
     }
 
+    // gets service members (e.g. helper bots) for exclusion
+    private getFunctionalMembers(): string[] {
+        const mFunctionalMembers = this.currentState.getStateEvents(UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name, "");
+        if (Array.isArray(mFunctionalMembers?.getContent().service_members)) {
+            return mFunctionalMembers!.getContent().service_members;
+        }
+        return [];
+    }
+
+    private getInvitedAndJoinedFunctionalMemberCount(functionalMembers: String[]): number {
+        return functionalMembers.reduce((count, m) => {
+            const membership = this.getMembers().find((member) => member.userId === m)?.membership;
+            if (membership && ["join", "invite"].includes(membership)) {
+                return count + 1;
+            }
+            return count;
+        }, 0);
+    }
```

Comment:
This is a really awkward interface: you have to *pass in* the list of functional members for it to filter. It should be called `getNumberOfInvitedAndJoinedMembersWhoseIDsMatchThisRandomList`

Instead: given it is only used in one place, inline it?